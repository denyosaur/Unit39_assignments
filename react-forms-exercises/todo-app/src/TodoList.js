import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import uuid from 'react-uuid';

const TodoList = () => {
    const [todos, setTodos] = useState([{ id: uuid(), task: "eat food" }]);

    const createTodo = (newTodo) => {
        setTodos([...todos, { id: uuid(), task: newTodo }]);
    };
    const deleteTodo = (id) => {
        setTodos(todos.filter(todo => {
            return todo.id !== id
        }));
    };

    return <div className="TodoList">
        <div><NewTodoForm createTodo={createTodo} /></div>
        <div>
            {todos.map(todo => {
                return <Todo task={todo.task} deleteTodo={() => deleteTodo(todo.id)} key={todo.id} />
            })}
        </div>
    </div>
}

export default TodoList;