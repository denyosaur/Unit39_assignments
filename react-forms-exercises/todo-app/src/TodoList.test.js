import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

//smoke test
test("TodoList should render", () => {
    render(<TodoList />)
})

//snapshot test
test("TodoList should render", () => {
    const { asFragment } = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();
})

//render form and test box
test("render form", () => {
    const { getByPlaceholderText } = render(<TodoList />);
    const taskInput = getByPlaceholderText("Enter Task Here");

    expect(taskInput).toBeInTheDocument();
});

test("render test box", () => {
    const { getByText } = render(<TodoList />);
    const btn = getByText("X");

    expect(btn).toBeInTheDocument();

    fireEvent.click(btn);
    expect(btn).not.toBeInTheDocument();
});
