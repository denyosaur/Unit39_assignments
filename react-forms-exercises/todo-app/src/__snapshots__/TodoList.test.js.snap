// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div>
      <div
        class="NewTodoForm"
      >
        <form>
          <label
            for="task"
          >
            Task: 
          </label>
          <input
            name="task"
            placeholder="Enter Task Here"
            type="text"
            value=""
          />
          <button>
            Submit
          </button>
        </form>
      </div>
    </div>
    <div>
      <div
        class="Todo"
      >
        <span
          class="Todo-task"
        >
          eat food
        </span>
        <button
          class="Todo-delete"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
